File: transcript-project/backend/src/TranscriptAnalyzer.Domain/Entities/HumanTask.cs
using TranscriptAnalyzer.Domain.Common;
using TranscriptAnalyzer.Domain.Enums;

namespace TranscriptAnalyzer.Domain.Entities;

public class HumanTask : TenantEntity
{
    public Guid CaseWorkflowId { get; private set; }
    public Guid StepExecutionId { get; private set; }
    public Guid? AssignedToUserId { get; private set; }
    public string Title { get; private set; }
    public string? Description { get; private set; }
    public DateTime? DueDate { get; private set; }
    public HumanTaskStatus Status { get; private set; }
    public DateTime? CompletedAt { get; private set; }
    public Guid? CompletedByUserId { get; private set; }
    public string? Decision { get; private set; }
    public string? Notes { get; private set; }

    public CaseWorkflow CaseWorkflow { get; private set; } = null!;
    public StepExecution StepExecution { get; private set; } = null!;

#pragma warning disable CS8618 // Required for EF Core
    private HumanTask() { }
#pragma warning restore CS8618

    public HumanTask(
        Guid organizationId,
        Guid caseWorkflowId,
        Guid stepExecutionId,
        string title,
        string? description = null,
        Guid? assignedToUserId = null,
        DateTime? dueDate = null) : base(organizationId)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(title);

        CaseWorkflowId = caseWorkflowId;
        StepExecutionId = stepExecutionId;
        Title = title.Trim();
        Description = description?.Trim();
        AssignedToUserId = assignedToUserId;
        DueDate = dueDate;
        Status = HumanTaskStatus.Pending;
    }

    public void StartWork()
    {
        Status = HumanTaskStatus.InProgress;
        SetUpdatedAt();
    }

    public void Complete(Guid completedByUserId, string? decision = null, string? notes = null)
    {
        Status = HumanTaskStatus.Completed;
        CompletedAt = DateTime.UtcNow;
        CompletedByUserId = completedByUserId;
        Decision = decision?.Trim();
        Notes = notes?.Trim();
        SetUpdatedAt();
    }

    public void Reassign(Guid newAssigneeUserId)
    {
        AssignedToUserId = newAssigneeUserId;
        Status = HumanTaskStatus.Reassigned;
        SetUpdatedAt();
    }

    public void Escalate()
    {
        Status = HumanTaskStatus.Escalated;
        SetUpdatedAt();
    }
}

File: transcript-project/frontend/src/lib/types/humanTask.ts
export type HumanTaskStatus =
  | 'Pending'
  | 'InProgress'
  | 'Completed'
  | 'Reassigned'
  | 'Escalated';

export interface HumanTaskListItem {
  id: string;
  caseWorkflowId: string;
  stepExecutionId: string;
  assignedToUserId?: string;
  title: string;
  dueDate?: string;
  status: HumanTaskStatus;
  completedAt?: string;
  decision?: string;
  createdAt: string;
}

export interface HumanTaskDetail {
  id: string;
  caseWorkflowId: string;
  stepExecutionId: string;
  assignedToUserId?: string;
  title: string;
  description?: string;
  dueDate?: string;
  status: HumanTaskStatus;
  completedAt?: string;
  completedByUserId?: string;
  decision?: string;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CompleteHumanTaskRequest {
  decision?: string;
  notes?: string;
}

export interface ReassignHumanTaskRequest {
  assignedToUserId: string;
}

export interface ListHumanTasksParams {
  page?: number;
  pageSize?: number;
  status?: HumanTaskStatus;
  assignedToUserId?: string;
  sortBy?: 'createdAt' | 'status' | 'dueDate' | 'title';
  sortOrder?: 'asc' | 'desc';
}

export interface PaginatedResponse<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}
